### PR TITLE
Add newArchitecture to README

### PR DESCRIPTION
**Pull request by sAleksovski**
# 📝 Why & how

There is a `newArchitecture` property defined in `react-native-libraries.schema.json`, and is already used by some libraries, but it was not documented in `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,10 +102,12 @@
   **(boolean)** - signify that a library is a development tool or is only a part of development process.
 - #### `template`
   **(boolean)** - signify that a library is a new project template.
+- #### `newArchitecture`
+  **(boolean)** - signify that a library supports the new architecture
```
Comment (Simek) [inline README.md:106]:
```suggestion
  **(boolean)** - signify that a library supports the new architecture

  > Set this tag only when automatic architecture detection fails for your package, despite it supports the new architecture.
```

Let's mention briefly the tag/field purpose.

**Comment by Simek** [comment]
For the context: The reason is that this field is only a temporary solution, workaround for the cases when automatic detection do not work, but there are plans in core team to standardize and formalize the `package.json` fields for RN, which in the future allows us to fully relay on automatic detection and remove this field.

But I guess we can list it in Readme, for now. I have made a suggestion which adds the explanation of the purpose of this field.

**Comment by sAleksovski** [comment]
> For the context: The reason is that this field is only a temporary solution, workaround for the cases when automatic detection do not work, but there are plans in core team to standardize and formalize the `package.json` fields for RN, which in the future allows us to fully relay on automatic detection and remove this field.
> 
> But I guess we can list it in Readme, for now. I have made a suggestion which adds the explanation of the purpose of this field.

I didn't know that, just saw that it was missing in the README.
You can decline the PR if you think it's better for it not to be included.

**Review by Simek — APPROVED** [review]
Let's add this field for now, thank you for the review update, LGTM! 👍

**Pull request merged**